conans/model/options.py

```diff
@@ -413,7 +414,7 @@ def values(self, vals):
             self._check_field(name)
             self._data[name].value = value
 
-    def propagate_upstream(self, package_values, down_ref, own_ref, output):
+    def propagate_upstream(self, package_values, down_ref, own_ref, output, ignore_unknown=False):
```

Comment:
I don't get the idea of the `ignore_unkown`, some code comments would be appreciated.

Reply:
Added docstring in PR: https://github.com/SSE4/conan/pull/4